Sources/StreamChat/Database/DTOs/ThreadDTO.swift

```diff
@@ -174,4 +210,10 @@ extension NSManagedObjectContext {
 
         return threadDTO
     }
+
+    func deleteAllThreads() throws {
+        let fetchRequest: NSFetchRequest<ThreadDTO> = NSFetchRequest(entityName: ThreadDTO.entityName)
```

Comment:
`NSBatchDeleteRequest` could be an option here, but since we do not expect to have so many threads, so this is also OK. Also batch delete needs extra code for contexts.

Reply:
Yeah, my first approach was with `NSBatchDeleteRequest`, and even used mergeChanges etc.. but I whenever I opened the app (Not fresh login) I would get duplicated threads, and I could not figure out why